docs/framework.md

```diff
@@ -143,3 +144,22 @@ class MyApplication {
 ```
 
 If you are using League\Container (recommended), then you may simply add and share your own classes to the same container.  If you are using some other DI container, then you should use [delegate lookup](https://github.com/container-interop/fig-standards/blob/master/proposed/container.md#14-additional-feature-delegate-lookup) to combine them.
+
+## Using a Custom Configuration Loader
+
+Robo provides a very simple configuration loader. If you wish to use more capable loader, you may opt to do so. Replace the call to `Robo::createConfiguration()` with code similar to the following:
+```
+use Robo\Config\Config;
+use Robo\Config\YamlConfigLoader;
+use Robo\Config\ConfigProcessor;
+
+$config = new Config();
+$loader = new YamlConfigLoader();
+$processor = new ConfigProcessor();
+$processor->extend($loader->load('defaults.yml'));
```

Comment:
Maybe `$config->extend($processor);` should be allowed. Note that `Config::extend()` creates a processor, and adds its current state to it. I was ambivalent about adding this method, as it creates coupling between the Config object and the config processor. I should either take out `Config::extend()`, and force the loose-coupling shown in the example, or embrace it. n.b. The current code allows repeated calls of `$config->extend($loader->load('defaults.yml'));`. However, in that instance you're stuck with the default processor.

I like decoupled better, but am still undecided here. @DavertMik ?